src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,194 @@
+          if (updateCompleted !== null) {
+            return { ...el, completed: !el.completed };
+          }
+
+          if (updateTitle) {
+            return { ...el, title: updateTitle };
+          }
+        }
+
+        return el;
+      }));
+    } catch (error) {
+      setTextError(ErrorType.PATCH);
+    }
+  };
+
+  const handlerDeleteTodo = () => {
+    setIsLoading(true);
+    setSelectedTodoId(id);
+    deleteTodoOnServer();
+  };
+
+  const handlerToggleClick = () => {
+    setIsLoading(true);
+    setSelectedTodoId(id);
+    changedTodoOnServer(null, !completed, id)
+      .finally(() => {
+        setSelectedTodoId(null);
+        setIsLoading(false);
+      });
+  };
+
+  const handlerSubmitNewTitle = () => {
+    if (!newTitle) {
+      handlerDeleteTodo();
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (newTitle === title) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    setIsEditing(false);
+    setIsLoading(true);
+    setSelectedTodoId(id);
+    changedTodoOnServer(newTitle, null, id)
+      .finally(() => {
+        setSelectedTodoId(null);
+        setIsLoading(false);
+        setSelectedTodoId(null);
+      });
+  };
+
+  useEffect(() => {
+    if (newTitleField.current) {
+      newTitleField.current.focus();
+    }
+  });
+
+  return (
+    <li
+      className={classNames(
+        'todo-item',
+        { 'todo-item--completed': completed },
+        { 'todo-item--editing': isEditing },
+      )}
+      onDoubleClick={() => setIsEditing(true)}
+    >
+      <div className="todo-item__view">
+        <input
+          type="checkbox"
+          className={classNames(
+            'todo-item__toggle',
+            { 'todo-item__toggle--completed': completed },
+          )}
+          onClick={handlerToggleClick}
+        />
+        <label className="todo-item__text">
+          {title}
+        </label>
+        <button
+          type="button"
+          className="todo-item__destroy"
+          data-cy="deleteTodo"
+          onClick={handlerDeleteTodo}
+          aria-label="Delete Todo"
+        />
+      </div>
+      {isEditing && (
+        <form onSubmit={(event) => {
+          event.preventDefault();
+          handlerSubmitNewTitle();
+        }}
+        >
+          <input
+            ref={newTitleField}
+            type="text"
+            className="todo-item__edit"
+            placeholder="Empty todo will be deleted"
+            value={newTitle}
+            onBlur={() => handlerSubmitNewTitle()}
+            onChange={({ target }) => {
+              setNewTitle(target.value);
+            }}
+            onKeyDown={({ key }) => {
+              switch (key) {
+                case 'Enter':
+                  handlerSubmitNewTitle();
+                  break;
+                case 'Escape':
+                  setIsEditing(false);
+                  break;
+
+                default:
+                  break;
+              }
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)